apps/emqx_connector/src/emqx_connector_http.erl

```diff
@@ -490,6 +490,48 @@ preproc_headers(Headers) when is_list(Headers) ->
         Headers
     ).
 
+wrap_auth_header(Headers) ->
+    lists:map(fun maybe_wrap_auth_header/1, Headers).
+
+maybe_wrap_auth_header({[{str, Key}] = StrKey, Val}) ->
+    {_, MaybeWrapped} = maybe_wrap_auth_header({Key, Val}),
+    {StrKey, MaybeWrapped};
+maybe_wrap_auth_header({_Key, Val} = Header) when is_function(Val) ->
+    %% Already wrapped?
+    Header;
+maybe_wrap_auth_header({Key, Val} = Header) when
+    is_binary(Key), (size(Key) =:= 19 orelse size(Key) =:= 13)
+->
+    %% We check the size of potential keys in the guard above and consider only
+    %% those that match the number of characters of either "Authorization" or
+    %% "Proxy-Authorization".
+    case try_bin_to_lower(Key) of
+        <<"authorization">> ->
+            {Key, maybe_wrap(Val)};
+        <<"proxy-authorization">> ->
+            {Key, maybe_wrap(Val)};
+        _Other ->
+            Header
+    end;
+maybe_wrap_auth_header(Header) ->
+    Header.
+
+try_bin_to_lower(Bin) ->
+    try iolist_to_binary(string:lowercase(Bin)) of
+        LowercaseBin -> LowercaseBin
+    catch
+        _:_ -> Bin
+    end.
+
+maybe_wrap([{str, Val}]) ->
+    [{str, emqx_secret:wrap(Val)}];
```

Comment:
we are passing `{str, function()}` to `emqx_placeholder:proc_tmpl`, 
this violates the spec of `tmpl_token()`.
maybe update the spec (and add a comment there).

Reply:
Good catch

Reply:
Turns out, this was actually a bug. We need to unwrap before calling proc_tmpl. Unlike what I believed no test uses auth-headers.